docs/source/pipelines.rst

```diff
@@ -4,194 +4,321 @@
 
 .. _steps-key-concept:
 
-Steps
-~~~~~
-
-Steps represent individual modeling or MLOps procedures, such as ingesting data,
-fitting an estimator, evaluating a model against test data, or deploying a model for
-real-time scoring. Steps accept a collection of well-defined inputs and produce
-well-defined outputs according to user-defined configurations and code. MLflow Pipelines
-intelligently caches step outputs, ensuring that steps are only executed if their
-inputs, code, or configurations have changed, or if such changes have occurred in dependent steps.
-Steps are composed to form :ref:`Pipelines <pipelines-key-concept>`.
+- **Steps**: A Step represents an individual ML operation, such as ingesting data, fitting an
+  estimator, evaluating a model against test data, or deploying a model for real-time scoring.
+  Each Step accepts a collection of well-defined inputs and produce well-defined outputs according
+  to user-defined configurations and code.
 
-.. code-block:: python
-  :caption: Example Python code defining an estimator that is fit during the |train step|
-            of the :ref:`MLflow Regression Pipeline <mlflow-regression-pipeline>`
+.. _pipelines-key-concept:
 
-  def estimator_fn():
-    from sklearn.linear_model import SGDRegressor
+- **Pipelines**: A Pipeline is an ordered composition of :ref:`Steps <steps-key-concept>` used to
+  solve an ML problem or perform an MLOps task, such as developing a regression model or performing
+  batch model scoring on production data. MLflow Pipelines provides
+  :py:class:`APIs <mlflow.pipelines.Pipeline>` and a :ref:`CLI <cli>` for running pipelines and
+  inspecting their results.
 
-    return SGDRegressor(random_state=42)
+.. _pipeline-templates-key-concept:
 
-.. code-block:: yaml
-  :caption: Example YAML configuration defining train / validation / test dataset split
-            ratios for the |split step| of the :ref:`MLflow Regression Pipeline
-            <mlflow-regression-pipeline>`.  This configuration demonstrates use of
-            |Jinja2| templating syntax to define a dynamic value for the ``split_ratios``
-            attribute that can be overriden by
-            :ref:`Pipeline Profiles <profiles-key-concept>`.
-
-  steps:
-    split:
-      # Train/validation/test split ratios
-      split_ratios: {{SPLIT_RATIOS|default([0.75, 0.125, 0.125])}}
+- **Templates**: A Pipeline Template is a git repository with a :ref:`standardized, modular layout
+  <pipeline-template-structure>` containing all of the customizable code and configurations for a
+  :ref:`Pipeline <pipelines-key-concept>`. Each template also defines its requirements, data science
+  notebooks, and tests. MLflow Pipelines includes :ref:`predefined templates <pipeline-templates>`
+  for a variety of model development and MLOps tasks.
 
-.. _pipelines-key-concept:
+.. _profiles-key-concept:
 
-Pipelines
-~~~~~~~~~
+- **Profiles**: Profiles contain user-specific or environment-specific configurations for a
```

Comment:
We haven't mentioned `pipeline.yaml` yet. Maybe we can say `pipeline.yam and profiles` here.

Reply:
I don't think `pipeline.yaml` is a separate concept. It's part of the template concept above. I've added `pipeline.yaml` to the template concept description; let me know if that works